webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -87,7 +91,12 @@ def update_correlation_plot(
             if selected_response not in valid_responses:
                 continue
             if df_index is None:
-                df_index = corrdf[selected_response].abs().sort_values().index.copy()
+                if sort_parameters:
+                    df_index = (
+                        corrdf[selected_response].abs().sort_values().index.copy()
+                    )
+                else:
+                    df_index = corrdf[selected_response].abs().index.copy()
```

Comment:
Would it be possible to isolate this functionality in a new function and test it?

Reply:
you know, that's a very good question. `df_index` only gets used in line 100, to reindex the existing correlation data frame. and perhaps (I'm not sure) the reindexing existed in the code only because the data was getting sorted! so if we don't sort, we don't need to reindex, and we don't need `df_index` at all!
gonna try this out : )

not sure about the own function and testing it - i'm gonna think a bit about that one, see that I can pull something out and write a meaningful test :sweat_smile:

Reply:
it seems like you were right - we only needed the index when we have to sort. i've adjusted the code accordingly.

about the test and refactoring - I don't know if this is what you meant, or if it adds much value, but the last commit extracts the part of the code responsible for sorting the dataframe, and i've added a test for testing said function. let me know what you think : )